resources/views/blade/form.blade.php

```diff
@@ -0,0 +1,19 @@
+@props([
+    'route',
+    'method' => 'POST',
+    'files' => false,
+])
+
+<form
+    action="{{ $route }}"
+    method="POST"
```

Comment:
Should this be `$method` **if** the $method was `GET` or `POST` - or just always the `$method`? (I believe browsers will turn anything other than `GET` into, effectively, a `POST` - but don't quote me on that).

Reply:
@uberbrady Ah interesting point.

>I believe browsers will turn anything other than GET into, effectively, a POST - but don't quote me on that

That is my understanding as well.

You probably saw that `$method` is used on line 16 since Laravel checks for `_method` in payloads but I hadn't accounted for any of our forms explicitly submitting GET. I believe that's overall not a great idea and skips Laravel's csrf protection.

I'm ok amending this but I don't know if we'll actually need/want it?

Reply:
Off the top of my head, I don't that any of our forms do - though maybe some search forms might? I'm also pretty sure none of our forms use `PATCH` nor `PUT` nor `DELETE` - but I suppose they could, and I at least like having the option. You also could (and probably should) pull any `@csrf` tags on `GET` forms, as they aren't respected there anyways.

Reply:
In general, if it's a GET (outside of search requests) it shouldn't be in a form at all. Forms - outside of searches - modify data, and no data should be modified via GET request ever. We don't pass `PUT` or `PATCH` in any forms we submit. The UI at least doesn't really care, it's more for the benefit of the API. You'll see a few places where we detect what's happening (save a few form requests - `CustomFieldRequest` and `SaveUserRequest`) because it's a POST, and it either exists or it doesn't.

Reply:
I agree with you on not using GETs for forms and I see what triggered me adding that bit of code. [A form involved uses `DELETE`](https://github.com/snipe/snipe-it/pull/15993/files#diff-b88fca117566bd1e45045bbf8a637a201daa3d21b166ff61d85afd0714a42ce6).

How'd you like me to approach this? Should I completely remove the component and inline the html or remove the ability to use `PUT` and `PATCH` in the component?

Reply:
IIRC, the only place where we use the `method_field` as `DELETE` is in the generic modal. Since it exists in only one place, my personal preference is to just use HTML throughout and make sure we keep that `{{ method_field('DELETE') }}` intact in the modal, but I'm open to discussion.

Reply:
No problem. I'll switch this to inline html. I'll move it to draft for now but might replace it with smaller PRs since it'll be easier to test.